geps/gep-1651.md

```diff
@@ -0,0 +1,274 @@
+# GEP-1651: Gateway Reachability
+
+* Issue: [#1651](https://github.com/kubernetes-sigs/gateway-api/issues/1651)
+* Status: Provisional
+
+(See status definitions [here](overview.md#status).)
+
+## TLDR
+
+Allow users to configure a GatewayAddress so that it is only reachable within
+a specific scope (ie. public/cluster local)
+
+## Goals
+
+- Cluster local gateways can be created in a declarative way (`cluster`) and
+  is accessible within a the bounds of the cluster's CNI
+- Define a `public` scope for addresses that are accessible to the internet
+- Implementations can define & support their own scopes
+
+## Non-Goals
+
+- Per-request/route scope
+- Not a lightweight service mesh
+
+## Introduction
+
+One of the early feature requests for Knative was the ability to deploy an
+application using Knative's HTTP routing support, but make it only available
+within the cluster. I want to be able to specify both the "internal"
+(service.namespace.svc) and "external" (service.namespace.example.com).
+Gateways using the same GatewayClass on the cluster, but ensure that the
+"internal" service is only reachable within the cluster. This would greatly
+simplify deployment for users over the instructions we have today.
+
+The reason why we require an "internal" Gateway is so we can take advantage of
+layer 7 load balancing for our internal services.
+
+## API
+
+We propose adding a new `reachability` field under the `spec.infrastructure` stanza of a Gateway.
+
+### Predefined Reachability Values
+
+Implementations MAY implement the following values for 'reachability' and MUST abide by
+their defined semantics.
+
+Value | Semantic
+-|-
+`public`|The address is reachable over the public internet
```

Comment:
Maybe just using "routable" instead of "reachable" is a better way to say this then?

Reply:
Works for me

Reply:
I agree with the definitions of these categories, I don't love the terms but also don't have good alternatives. When we have two levels, they largely make sense, but it's hard to come up with 3 levels that are all clearly complementary.

For example a word like "Internal" is inherently relative to _something_. It could be relative to a cluster or a network or something else entirely. One alternative would be to use something like "VPC" that is more analagous to "Cluster", but also somewhat imprecise when things like VPC-peering exist.

One more concrete suggestion, it seems like the words we use to represent the non-cluster concepts should be antonyms, so either "External" and "Internal" or "Public" and "Private". 

Welcome to better terms here, but haven't been able to come up with any better suggestions.

Reply:
+1 to antonyms, I think that `public` and `private` is probably better than `external` and `internal`.

Reply:
Updated the GEP based on this - please take a look again.